### ISSUE TITLE
[Feature Request] [L] Support for Android 11

### ISSUE THREAD

**Issue by bgavrilMS** [AzureAD/microsoft-authentication-library-for-dotnet#2139]
# Android 11 impact

•	Unable to query package manager for and launch browsers that support custom tabs
•	Unable to query package manager for and launch regular browsers that don’t support custom tabs
•	Unable to query package manager for Broker apps installed on the device (We could still see the Authenticators on the device but when trying to query them via Package Manager it would not be found)

It is important to note that these issues are only observed **when the app is targeting SDK 30 & running on Android 11 device**

## Fix 1 

According to Android team, a library can have its own manifest. As far as I know, MSAL.NET does not use this mechanism. To mitigate Android 11 impact, we can update our own manifest, similar to: 

https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1095/files

This still requires customers to update to the latest version of MSAL.

## Fix 2 

Update our docs with app manifest requirements as per:

https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1095/files

**Comment by tipa** [comment]
I tried to implement "Fix 2" but the app still launches the browser instead of a custom tab.
Is the fix outdated or anything else to pay attention to?

**Comment by tipa** [comment]
Problem is still present with v2.25.0 - cannot use Custom Tabs with Android 29+ - the login flow always opens the browser.
Would it be possible to at least change this code here?
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/d451b16a9ae66742047b0ba37fb74d875a69056d/src/client/Microsoft.Identity.Client/Platforms/Android/SystemWebview/AuthenticationActivity.cs#L37-L45
The comment itself says that "Must use the classic support" but for Android 29+ the wrong(?) Intent Action is used.
I think it should be `android.support.customtabs...` and not `androidx.browser.customtabs...`

**Comment by jmprieur** [comment]
@henrik-me @trwalke @bgavrilMS @jennyf19 
Do we want to take this this sprint?

**Comment by tipa** [comment]
Regarding proposed "Fix 1" there has been some talk here about how it would be possible to include the extra stuff to the AndroidManifest.xml: https://github.com/xamarin/Essentials/issues/1392#issuecomment-762865959

**Comment by tipa** [comment]
I still have to use v4.23 on Android because of this issue. Any chance it will be fixed in the next update?

**Comment by bgavrilMS** [comment]
@trwalke - can you take a look at this for the next MSAL release? We should not block people on older SDK versions. Also good to better understand differences between Android 10 and 11.

**Comment by bgavrilMS** [comment]
Looks like 2 issues are present here, @trwalke - please split it up if needed.

**Comment by bgavrilMS** [comment]
@trwalke - given that we need big manifest changes, I wonder if it won't be faster to add the manifest.xml directly to MSAL, similar to what MSAL.Android do ?

 This would avoid us having to change all the samples, wiki and MSDN?

**Comment by trwalke** [comment]
@bgavrilMS in my previous attempt I could not get the manifest to work when it was attached to MSAL. it is easier for android as they have a project designed to accept manifest files. I will take another look at it to see if i can get it to work but I may have to reach out to the Xamarin guys

**Comment by trwalke** [comment]
Cannot merge PR until next week due to restrictions in MS Docs. Blocking until then

**Comment by trwalke** [comment]
@tipa The issue you mention here seems to be related to version 29 as well so I am splitting the issue into 2.
The second issue with the custom tab support not working will be tracked here https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2418

**Comment by tipa** [comment]
This is still not working for me after 4.28, even if it says so in the changelog.
I assume it's because MSAL still queries for the wrong package, as I already suspected [here](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2139#issuecomment-764492298)
(`androidx.browser.customtabs.action.CustomTabsService` instead of `android.support.customtabs.action.CustomTabsService`)
Looks like I have to stick with v4.23 even longer...

**Comment by bgavrilMS** [comment]
CC @trwalke

**Comment by trwalke** [comment]
@tipa This issue is only for adding android 11 support. The issue you mentioned is related to Android 29+ which is android 10 and 11 and is about custom tabs so it does not exactly relate to this issue. I moved your issue to a separate issue here #2418

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2139#issuecomment-786769878

**Comment by trwalke** [comment]
Information on how to implement Android 11 support can be found here:
https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-net-xamarin-android-considerations#android-11-support

**Comment by tipa** [comment]
What exactly does "Android 11 support" mean?
I was able to use MSAL / CustomTabs without issues with v4.23 - but I cannot them with v4.28. Whenever I try, it opens the system browser in a new task (meaning it exits the app).
Your linked docs mention added this part:
```
  <intent>
    <action android:name="android.support.customtabs.action.CustomTabsService" />
  </intent>
```
But why?
As I pointed out, a different package is being queried so this shouldn't make any difference.
Or what am I missing here?

**Comment by trwalke** [comment]
@tipa 
"Android 11 support" means enabling customers to use our product on Android 11 by following our guidance. This has nothing to do with the issue you are referring to. As I said earlier, the issue you have is being tracked in a separate bug here (https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2418). Please see my previous comments

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2139#issuecomment-802196636
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2139#issuecomment-786769878

**Comment by bgavrilMS** [comment]
@trwalke - not having system browser support on Android 11 is pretty big gap that needs to be addressed, and we should mention it in the docs.
